api/include/opentelemetry/plugin/tracer.h

```diff
@@ -48,7 +48,7 @@ class Span final : public trace::Span
 
   bool IsRecording() const noexcept override { return span_->IsRecording(); }
 
-  trace::Tracer &tracer() const noexcept override { return *tracer_; }
```

Comment:
For the tracer issue I kept it because I saw them being added fairly recently and they might have a point on this. And I think they could help eventually getting this to work here. I think the removal of this methods hence is only a temporary one.